src/plasma/plasma_client.cc

```diff
@@ -465,28 +473,89 @@ void plasma_contains(PlasmaConnection *conn, ObjectID obj_id, int *has_object) {
   }
 }
 
+static void compute_block_hash(const unsigned char *data,
+                               int64_t nbytes,
+                               uint64_t *hash) {
+  XXH64_state_t hash_state;
+  XXH64_reset(&hash_state, XXH64_DEFAULT_SEED);
+  XXH64_update(&hash_state, data, nbytes);
+  *hash = XXH64_digest(&hash_state);
+}
+
+static inline bool compute_object_hash_parallel(XXH64_state_t *hash_state,
+                                                const unsigned char *data,
+                                                int64_t nbytes) {
+  const uint64_t numthreads = THREADPOOL_SIZE;
+  uint64_t threadhash[numthreads + 2];
+  const uint64_t block_size = BLOCK_SIZE;
+  /* Calculate the first and last aligned positions in the data stream. */
+  const uint64_t data_address = reinterpret_cast<uint64_t>(data);
+  uint64_t left_address = (data_address + block_size - 1) & ~(block_size - 1);
```

Comment:
I think it should be possible to simplify a bunch of the code here, e.g., `left_address = data_address`. Is that right?